### ISSUE TITLE
Issue still happening

### ISSUE THREAD

**Issue by phuqertrfc** [jamesbrindle/YTMusicUploader#12]
Hi, 

I raised an issue earlier in the week that has been closed today.

The issue was as follows:

I am also having this same issue.
I transferred from Play Music a while back but noticed not everything had transferred, plus I didn't like the fact I had to manually add new music.
So I've installed this app and ran it for the first time.
It discovered 34801 files and started to go through them all. Most of them showed as already present but a few that were missing got added.
It was running really well but then when it got to 20532 uploaded it just stopped and hung at "Gathering uploaded artists"
I left it for hours but nothing.
I restarted but just hung again.
I clicked on "Reset uploaded music states" and it started from 0 again, but then it hung at exactly 20532 uploads for the second time.
I did wonder if it was an issue with the remaining files, but the second time I ran it, it processed different files as it was still finding stuff that hadn't been uploaded, so it seems to be an issue with this tool rather than the files.
Seems strange that it hangs at exactly the same number each time though.
I tried a third time and exactly the same result.

I also tried adding some new music to the folder to see if it picked that up and the discovered files increased but the uploads still hung at 20532.

I've now installed the updated version, however the same issue seems to be happening but now it is now stuck at 18915 of 34819 and has now been stuck at Gathering Uploaded Artists for the last 2 days.

Any ideas?

Cheers
Steve

**Comment by jamesbrindle** [comment]
Oh no!

Can you confirm a couple of things.

1) Have you noticed your CPU going crazing when it's stuck at 'Gathering Uploaded Artists'? I fixed a pretty obvious loop bug just yesterday and was pretty confident this would fix the issue. It would have only happened when it hit a file larger than 300 MB. If not, it's perhaps something else I overlooked.

2) Have you checked the Application log (the new icon on the top left that says 'Log') and noticed any particular errors?

3) When did you install the updated version? I only updated it yesterday, and then pushed another version tonight with a couple more minor fixes.

Regards,

J

**Comment by jamesbrindle** [comment]
Hey,

I've pushed another version here: https://github.com/jamesbrindle/YTMusicUploader/releases/tag/v1.5.2

I found a good few bugs this time while digging into the code after not seeing it for a few weeks... Typically trying to solve one issue while creating another. Lets hope this version is better.

You say you're stuck on "Gathering uploaded artists". This is strange, because I did fix the bug that definitely caused this kind of issue. However, it's possible something else was happing while trying to recursively get your Uploads collection from YouTube Music, and causing the responses to get stuck... Therefore I've implemented a timeout of 6 minutes to get all the responses back... If you encounter a 'timeout' message box, please let me know, because I'll have then discovered where the issue is occurring... if not, I'll have to keep digging.

Regards,

J

**Comment by phuqertrfc** [comment]
Cheers, really appreciate you looking into this. Won't get a chance to install the new version until later this week, but will come back to you and let you know how I get on.

Cheers
Steve

**Comment by phuqertrfc** [comment]
Quick update....

That's worked a treat.
Thank you very much for looking into that and fixing it.

Don't suppose you can now recode the YTMusic app itself and change the views so that you can view albums by Artist rather than just listing all the songs :-)

Cheers
Steve

**Comment by phuqertrfc** [comment]
One other question before you close this....
The way I managed my music previously was via iTunes on my laptop, so any new CDs I bought (I'm that generation that still love physical copies :-)) would get imported into iTunes and the Google Play Music background app would pick up that the itunes folder had been updated and would import the new music and it would then be available to me on my android phone via the app.
This is essentially how I'm now setup with YTMusic thanks to your uploader app, but one thing I'm sure the google app did was recognise new playlists in iTunes and uploaded them. 
Does your app do this or is it something it could potentially do?

Cheers
Steve
